src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -700,7 +700,7 @@ private BoundExpression CreateAnonymousFunctionConversion(SyntaxNode syntax, Bou
             // UNDONE: is converted to a delegate that does not match. What to surface then?
 
             var unboundLambda = (UnboundLambda)source;
-            var boundLambda = unboundLambda.Bind((NamedTypeSymbol)destination, isExpressionTree: destination.IsGenericOrNonGenericExpressionType(out _));
+            var boundLambda = unboundLambda.Bind((NamedTypeSymbol)destination, isExpressionTree: destination.IsGenericOrNonGenericExpressionType(out _)).WithInAnonymousFunctionConversion();
```

Comment:
Would it be prohibitively difficult to pass down the information, or maybe set it once in "checked" fashion a la the WasCompilerGenerated flag?

Reply:
It wouldn't be difficult, but not sure what to do about caching in `UnboundLambda.Bind`:

https://github.com/dotnet/roslyn/blob/aecd49800750d64e08767836e2678ffa62a4647f/src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs#L563-L574

`WasCompilerGenerated` flag is different in this regard - it's just copied from the unbound lambda.

From debugging tests, it seems to me that `CreateAnonymousFunctionConversion` is called only once per lambda (unlike `UnboundLambda.Bind` which is called multiple times and it's not always known whether we have a successful conversion). From that I would say it doesn't make sense to add the version with `InAnonymousFunctionConversion: true` to the cache and keep the cloning?